tests/test_pqdict.py

```diff
@@ -4,467 +4,512 @@
+    pq3 = minpq(A=5, B=8, C=7, D=3, E=9, F=12, G=1)
+    assert pq0 == pq1 == pq2 == pq3
+
+
+def test_equality():
+    # eq
+    pq1 = pqdict(sample_items)
+    pq2 = pqdict(sample_items)
+    assert pq1 == pq2
+    assert not pq1 != pq2
+    # ne
+    pq2[random.choice(sample_keys)] += 1
+    assert not pq1 == pq2
+    assert pq1 != pq2
+    # pqdict == regular dict if they have same key/value pairs
+    adict = dict(sample_items)
+    assert pq1 == adict
+    # TODO: FIX?
+    # pqdicts evaluate as equal even if they have different
+    # key functions and/or precedence functions
+    pq3 = maxpq(sample_items)
+    assert pq1 == pq3
+
+
+def test_minpq():
+    pq = minpq(A=5, B=8, C=7, D=3, E=9, F=12, G=1)
+    assert list(pq.popvalues()), [1, 3, 5, 7, 8, 9, 12]
+    assert pq.precedes == operator.lt
+
+
+def test_maxpq():
+    pq = maxpq(A=5, B=8, C=7, D=3, E=9, F=12, G=1)
+    assert list(pq.popvalues()), [12, 9, 8, 7, 5, 3, 1]
+    assert pq.precedes == operator.gt
+
+
+def test_fromkeys():
+    # assign same value to all
+    seq = ["foo", "bar", "baz"]
+
+    pq = pqdict.fromkeys(seq, 10)
+    for k in seq:
+        assert pq[k] == 10
+
+    pq = pqdict.fromkeys(seq, float("inf"))
+    for k in seq:
+        assert pq[k] == float("inf")
+    pq["spam"] = 10
+    assert pq.pop() == "spam"
+
+    pq = pqdict.fromkeys(seq, float("-inf"), precedes=operator.gt)
+    for k in seq:
+        assert pq[k] == float("-inf")
+    pq["spam"] = 10
+    assert pq.pop() == "spam"
+
+
+# The next group of tests were originally in class TestDictAPI
+
+
+def test_len():
+    pq = pqdict()
+    assert len(pq) == 0
+    pq = pqdict(sample_items)
+    assert len(pq) == len(sample_items)
+
+
+def test_contains():
+    pq = pqdict(sample_items)
+    for key in sample_keys:
+        assert key in pq
+
+
+def test_getitem():
+    pq = pqdict(sample_items)
+    for key, value in sample_items:
+        assert pq[key] == value
+
+
+def test_setitem():
+    n = len(sample_items)
+    pq = pqdict(sample_items)
+    pq["new"] = 1.0  # add
+    assert pq["new"] == 1.0
+    assert len(pq) == n + 1
+    pq["new"] = 10.0  # update
+    assert pq["new"] == 10.0
+    assert len(pq) == n + 1
+
+
+def test_delitem():
+    n = len(sample_items)
+    pq = pqdict(sample_items)
+    key = random.choice(sample_keys)
+    del pq[key]
+    assert len(pq) == n - 1
+    assert key not in pq
+    with pytest.raises(KeyError):
+        pq.pop(key)
+
+
+def test_copy():
+    pq1 = pqdict(sample_items)
+    pq2 = pq1.copy()
+    assert pq1 == pq2
+    key = random.choice(sample_keys)
+    pq2[key] += 1
+    assert pq1[key] != pq2[key]
+    assert pq1 != pq2
+
+
+# inherited implementations
+def test_get():
+    pq = pqdict(sample_items)
+    assert pq.get("A") == 5
+    assert pq.get("A", None) == 5
+    assert pq.get("does_not_exist", None) is None
+
+    # INVESTIGATE: in unit test, this succeded
+    # with pytest.raises(KeyError):
+    #     pq.get('does_not_exist')
```

Comment:
I believe `assert pq.get("does_not_exist") is None` should pass. I think this behavior is sensible and is consistent with `dict`. If you agree, let's change to that.

Though, I have no idea why the KeyError test used to pass before...

Reply:
Yes, I agree. Should be changed in 38a5481